test/ered_SUITE.erl

```diff
@@ -192,6 +200,50 @@ t_command_pipeline(_) ->
     no_more_msgs().
 
 
+t_client_crash(_) ->
+    R = start_cluster(),
+    Port = get_master_from_key(R, <<"k">>),
+    Addr = {"127.0.0.1", Port},
+    AddrToPid0 = ered:get_addr_to_client_map(R),
+    Pid0 = maps:get(Addr, AddrToPid0),
+    monitor(process, Pid0),
+    TestPid = self(),
+    SleepCommand = [<<"DEBUG">>, <<"SLEEP">>, <<"3">>],
+    spawn_link(fun () ->
+                       Result = ered:command(R, SleepCommand, <<"k">>),
+                       TestPid ! {crashed_command_result, Result}
+               end),
+    ered:command_async(R, SleepCommand, <<"k">>,
+                       fun (Reply) ->
+                               %% We never get this reply. The ered_client
+                               %% process crashes before it is sent.
+                               TestPid ! {crashed_async_command_result, Reply}
```

Comment:
Looking at `ered_client:terminate/2`, I would expect the reply `{error, {client_stopped, crash}}` (sent by `reply_all`). I don't understand why it isn't sent. Since the ered_client traps exists, this should happen before the ered_client process dies...

Reply:
Hum.. an `exit(Pid0, crash)` should trigger a call to `ered_client:terminate/2`.
Is the `handle_cast(command...)` not called in order during a termination, i.e. have not yet added the Command to `State#st.waiting`?
weird..  but some printouts might show

Reply:
Right. A sleep of 100ms solved it. Added another test where the ered_client is hard killed to cover the other case (or at least a similar result).